src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift

```diff
@@ -9,17 +9,17 @@ extension TraceLocation {
 
 	// MARK: - Internal
 	
-	func qrCode(size: CGSize = CGSize(width: 400, height: 400)) -> UIImage? {
+	func qrCode(size: CGSize = CGSize(width: 400, height: 400), qrCodeErrorCorrectionLevel: String = "M") -> UIImage? {
```

Comment:
Can you Please use an enum for this? qrCodeErrorCorrectionLevel is already an enum inside the protobuf we should keep it like this, if you need the string maybe create a property inside the enum to return the corrosponding letter but i think an enum would be enough

Reply:
Yes, we need string letters; already present enums are meaningless for QR code generation, adding a property with in the enum of pb files will already change them, have to find an alternative.

Reply:
you can declare enums with string property and use the cases as string or at least their raw values

Reply:
or you could pass it as associated value in the enum case